extjsdk/src/test/java/io/vantiq/extjsdk/TestExtensionWebSocketClient.java

```diff
@@ -83,8 +83,7 @@ public void testAuthenticateWithUsername() throws InterruptedException {
         client.webSocketFuture.complete(true);
         
         client.authenticate(user, pass);
-        // do twice because the initial attempt is asynchronous, so the data will not be ready for the asserts 
-        client.authenticate(user, pass);  
+        Thread.sleep(100); // Wait for asynchronous action to complete 
```

Comment:
At some point in the future, I'd add some capability to spin here for a "few" iterations.  On slower machines, different environments, etc., the 100/15ms will be unpredictable and folks won't understand what to do about failed tests.  So have something that will spin up to 5 or 10 seconds then fail.  But checking every 100ms is good as that keeps the tests short under good situations.